File: terangaguest_app/lib/models/leisure_category.dart
/// Catégorie principale (Sport ou Loisirs) avec ses activités enfants.
class LeisureMainCategoryDto {
  final int id;
  final String name;
  final String? description;
  final String type; // sport | loisirs
  final int displayOrder;
  final List<LeisureCategoryDto> children;

  LeisureMainCategoryDto({
    required this.id,
    required this.name,
    this.description,
    required this.type,
    required this.displayOrder,
    required this.children,
  });

  factory LeisureMainCategoryDto.fromJson(Map<String, dynamic> json) {
    final childrenList = json['children'] as List<dynamic>? ?? [];
    return LeisureMainCategoryDto(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      type: json['type'] as String? ?? 'sport',
      displayOrder: json['display_order'] as int? ?? 0,
      children: childrenList
          .map((e) => LeisureCategoryDto.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}

/// Sous-catégorie / activité (Golf & Tennis, Fitness, Spa, etc.).
class LeisureCategoryDto {
  final int id;
  final String name;
  final String? description;
  final String type; // golf_tennis, fitness, spa, other
  final int displayOrder;

  LeisureCategoryDto({
    required this.id,
    required this.name,
    this.description,
    required this.type,
    required this.displayOrder,
  });

  factory LeisureCategoryDto.fromJson(Map<String, dynamic> json) {
    return LeisureCategoryDto(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      type: json['type'] as String? ?? 'other',
      displayOrder: json['display_order'] as int? ?? 0,
    );
  }
}

File: app/Models/LeisureCategory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Models\Scopes\EnterpriseScopeTrait;
use Spatie\Translatable\HasTranslations;

class LeisureCategory extends Model
{
    use EnterpriseScopeTrait, HasTranslations;

    public array $translatable = ['name', 'description'];

    /** Catégories principales (écran 1) */
    public const TYPE_SPORT = 'sport';
    public const TYPE_LOISIRS = 'loisirs';

    /** Sous-catégories (écran 2) - type = slug pour l'app */
    public const TYPE_GOLF = 'golf';
    public const TYPE_TENNIS = 'tennis';
    public const TYPE_SPA = 'spa';
    public const TYPE_FITNESS = 'fitness';
    public const TYPE_OTHER = 'other';

    protected $fillable = ['enterprise_id', 'parent_id', 'name', 'description', 'type', 'display_order', 'is_active'];

    protected $casts = ['display_order' => 'integer', 'is_active' => 'boolean'];

    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    public function enterprise()
    {
        return $this->belongsTo(Enterprise::class);
    }

    public function parent()
    {
        return $this->belongsTo(LeisureCategory::class, 'parent_id');
    }

    public function children()
    {
        return $this->hasMany(LeisureCategory::class, 'parent_id');
    }

    public function scopeOrdered($query)
    {
        return $query->orderBy('display_order', 'asc')->orderBy('name', 'asc');
    }

    public function scopeTopLevel($query)
    {
        return $query->whereNull('parent_id');
    }

    public function getTypeLabelAttribute(): string
    {
        return match ($this->type) {
            self::TYPE_SPORT => 'Sport',
            self::TYPE_LOISIRS => 'Loisirs',
            self::TYPE_GOLF => 'Golf',
            self::TYPE_TENNIS => 'Tennis',
            self::TYPE_SPA => 'Spa & Wellness',
            self::TYPE_FITNESS => 'Sport & Fitness',
            self::TYPE_OTHER => 'Autre',
            default => ucfirst(str_replace('_', ' ', $this->type ?? '')),
        };
    }

    public function isMainCategory(): bool
    {
        return $this->parent_id === null;
    }
}
